animations/src/main/java/com/chiralbehaviors/CoRE/phantasm/model/PhantasmCRUD.java

```diff
@@ -130,69 +139,87 @@ public RuleForm addChild(RuleForm instance,
+                                RuleForm instance,
                                 XDomainNetworkAuthorization<?, ?> auth,
                                 @SuppressWarnings("rawtypes") List<ExistentialRuleform> children) {
         if (instance == null) {
             return null;
         }
         NetworkedModel<RuleForm, Network, ?, ?> networkedModel = model.getNetworkedModel(instance);
-        checkUPDATE(auth, networkedModel);
+        if (!checkUPDATE(facet, networkedModel)
+            || !checkUPDATE(auth, networkedModel)) {
+            return instance;
+        }
         NetworkAuthorization<?> childAuth = auth.isForward() ? resolveTo(auth,
                                                                          model)
                                                              : resolveFrom(auth,
                                                                            model);
-        for (@SuppressWarnings("rawtypes")
-        ExistentialRuleform child : children) {
-            checkREAD(child, model.getUnknownNetworkedModel(child));
-            if (childAuth.getClassification() instanceof Agency) {
-                networkedModel.authorize(instance, model.getEntityManager()
-                                                        .merge(auth.getConnection()),
-                                         (Agency) child);
-            } else if (childAuth.getClassification() instanceof Location) {
-                networkedModel.authorize(instance, model.getEntityManager()
-                                                        .merge(auth.getConnection()),
-                                         (Location) child);
-            } else if (childAuth.getClassification() instanceof Product) {
-                networkedModel.authorize(instance, model.getEntityManager()
-                                                        .merge(auth.getConnection()),
-                                         (Product) child);
-            } else if (childAuth.getClassification() instanceof Relationship) {
-                networkedModel.authorize(instance, model.getEntityManager()
-                                                        .merge(auth.getConnection()),
-                                         (Relationship) child);
-            } else {
-                throw new IllegalArgumentException(String.format("Invalid XdAuth %s -> %s",
-                                                                 instance.getClass()
-                                                                         .getSimpleName(),
-                                                                 childAuth.getClassification()
-                                                                          .getClass()
-                                                                          .getSimpleName()));
-            }
-        }
+        children.stream()
+                .filter(child -> checkREAD(child,
```

Comment:
In future versions, we should figure out a way to return errors on non-readable children.